### DISCUSSION TITLE
List of random issues

### DISCUSSION THREAD

**Discussion by Sqrcz** [themesberg/flowbite-vue#236] [Ideas]
Hey :wave: 

While refactoring this library to introduce component prefix and better linting I wen't through every single file... and since I was updating components as well as docs I have few thoughts about implementation, styling, usage of props, errors and the other more or less opinionated things...

Here is the "raw" list of issues I found during the above process... some would require separate tickets/discussions... some just decision which way to go with them.

When responding please use "Issue number" and if we'll create dedicated tickets... I'll edit this message to link them properly.

 1. ~~paragraph should have mb3 as default with mb-0 for the last one~~ - #257 
 2. ~~heading should be H element~~ - #199 
 3. link should be `<router-link>` if prop `to` is set
 4. images shouldn't be linked from other repos like `flowbite-svelte`
 5. no need for props for things that can be achieved by tailwind class (align, size)
 6. ~~review and remove `/* eslint-disable...`~~ - #376
 7. ~~avatar status is not visible when image is used~~ - #258 
 8. ~~`avatar-stack-counter` shouldn't be a link~~ - it is link in [Figma - Avatar - Stacked](https://flowbite.com/docs/components/avatar/#stacked)
 9. type check for `avatar-stack-counter` `total` doesn't work
 10. `breadcrumb-item` should be used internally and `breadcrumbs` component should accept array
 11. `breadcrumb-item` should be `<router-link>` if prop `to` is set
 12. ~~`<Accordion :open-first-item="false">` should be just `<Accordion first-item-closed>`~~ - #351 
 13. button loading state should replace content without affecting button size - this is aligned with [Flowbite - Button - Loader](https://flowbite.com/docs/components/avatar/#stacked) (worth escalating)
 14. carousel component needs refactoring, slide animation is not working, should have prop for `auto-start` and `interval` instead of `slide` and `slide-interval`, `animation` prop is not used
 16. ~~dropdown placements `right` and `left` are working on second click (bad position on first one)~~
 17. ~~should dropdown arrow be aligned with dropdown placement?~~
 18. `fwb-list-group-item` hover only makes sense if it's clickable (should accept `href` or `to`)
 19. `pagination` could use `button-group` internally
 20. ~~disabled pagination button should not have hover effect ("Previous" / "Next")~~ - #319 
 21. `besideText` slot in `fwb-rating` component should have default styling
 22. rating component should be able to manage floats
 23. it should be possible to have rating count without a link (it's called `review-text`)
 24. spinner size should not be based on fixed classes but something like: `w-${size} h-${size}`
 25. tabs `directive` should be boolean and named like `directive-show` (easier DX)
 26. `click:pane` should pass through event data
 27. Toast component `type="empty"` shouldn't be needed (`without-icon` would make more sense and allow for more flexibility)
 28. example for Interactive Toast doesn't work without `flowbite-themable` component
 29. Toasts are lacking one of the main functionality (dynamic dysplaying in predefined positions)
 30. "hovered" Tooltip dissapears on click
 31. ~~`file-input` should have `accept` prop to allow for file extensions filtering~~ - #304 
 32. ~~`file-input` with `deopzone="true"` doesn't accept multiple files~~ - #379 
 36. checkbox label could have class `select-none` so it's text is not selectable on multiple clicks
 37. ~~checkbox and radio lists should be created by dedicated components~~ - #368 
 39. ~~in `range` component label's `for` attribute can't be hard-coded~~
 40. table component should not require building table manually... (https://bootstrap-vue.org/docs/components/table is a good example)
 41. ~~Alert's additional-content example could use `fwb-button` instead of HTML one~~ - #377 
 43. we should unify sizes... sometime we have prop `size` with different options and sometimes we have dedicated prop like `large` (pagination)
 44. `Failed to resolve component: router-link` in Sidebar component (we need to add vue router dependency for that to work)
 45. `file upload` component button is too dark in light theme

**Comment by cogor** [comment]
2. Done (#225)

**Comment by Sqrcz** [comment]
1. redy for review in #257

**Comment by Sqrcz** [comment]
7. ready for review in #258

**Comment by Sqrcz** [comment]
20. Fixed in https://github.com/themesberg/flowbite-vue/pull/319

**Comment by wazeerc** [comment]
@Sqrcz Is this list of issues up-to-date?

**Reply by Sqrcz** [reply]
Hey @wazeerc 👋 

Few components were completely refactored... I'll go over this list this week and update it... thanks for the reminder ;)

**Reply by Sqrcz** [reply]
Hey, I've updated the list above... If you have time or want to... be happy to pick something ;)

**Comment by wazeerc** [comment]
36. Done - #377

**Comment by wazeerc** [comment]
32. Done - #378

**Comment by wazeerc** [comment]
31. Done - #379

**Comment by Sqrcz** [comment]
Thank you!

We have a long weekend in Poland now... I'll be able to take a look a those next week.
